interop/shared/src/main/scala/scalaz/zio/interop/future.scala

```diff
@@ -1,38 +1,64 @@
 package scalaz.zio
 package interop
 
+import scalaz.zio.ExitResult.Cause
+
 import scala.concurrent.{ ExecutionContext, Future }
 import scala.util.{ Failure, Success }
 
 object future {
 
   implicit class IOObjOps(private val ioObj: IO.type) extends AnyVal {
     def fromFuture[A](ftr: () => Future[A])(ec: ExecutionContext): IO[Throwable, A] =
-      IO.async { (cb: ExitResult[Throwable, A] => Unit) =>
-        ftr().onComplete {
-          case Success(a) => cb(ExitResult.succeeded(a))
-          case Failure(t) => cb(ExitResult.checked(t))
-        }(ec)
+      IO.suspend {
+        val f = ftr()
+        f.value match {
+          case Some(result) =>
+            result match {
```

Comment:
Using `fold` over pattern matching has become kind of the idiomatic style in ZIO, what do you think of
```scala
result.fold(t => IO.fail0(Cause.checked(t)), IO.now)
```